packages/babel-plugin-extract-export-names/index.js

```diff
@@ -0,0 +1,65 @@
+const {declare} = require('@babel/helper-plugin-utils')
+
+class BabelPluginExtractExportNames {
+  constructor() {
+    const names = []
+    this.state = {names}
+
+    this.plugin = declare(api => {
+      api.assertVersion(7)
+      const {types: t} = api
+
+      const handleDeclarations = node => {
+        const { declarations } = node.declaration || {}
+
+        if (!declarations) {
```

Comment:
Won’t this always be false? {} is truthy

Reply:
I should probably write this differently so it's more clear but it _should_ work since `declarations` is being destructured:

```js
> const {declarations} = null || {}
undefined
> declarations
undefined
```

Reply:
Addressed in https://github.com/mdx-js/mdx/pull/1160/commits/44e3e49270682357997bfa246a337247bd2bd960 to make it more clear.

Also fixed an incorrect handling of the default below which was incorrect 5f73b0b5b961c45ca170204e46fd8525a741851b.

Reply:
Ah, ya. It’s just a weird behavior to make use of. But I gotcha :)